From 08c7045c2f123a60b6d5d913daef7753cd53649d Mon Sep 17 00:00:00 2001
From: Paul Gardiner <mythtv@glidos.net>
Date: Sat, 26 Sep 2020 16:22:22 +0100
Subject: [PATCH] Fix incorrect artwork urls returned from ttvdb grabber

When performing a manual search for metadata for a video, the artwork
fetch was failing: no thumbnail was shown and no artwork was being
associated with the video. Ticket 13518 mentioned strange urls containing
"banners/_cache//banners". This commit fixes the bad urls and seems to
restore the downloading of the artwork. I don't know what led to the need
for this.
---
 .../bindings/python/MythTV/ttvdb/XSLT/tvdbQuery.xsl  | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/mythtv/bindings/python/MythTV/ttvdb/XSLT/tvdbQuery.xsl b/mythtv/bindings/python/MythTV/ttvdb/XSLT/tvdbQuery.xsl
index bf7ad06a82e..ee6c8ade816 100644
--- a/mythtv/bindings/python/MythTV/ttvdb/XSLT/tvdbQuery.xsl
+++ b/mythtv/bindings/python/MythTV/ttvdb/XSLT/tvdbQuery.xsl
@@ -48,22 +48,22 @@
                         <xsl:if test=".//poster/text() != ''">
                             <xsl:element name="image">
                                 <xsl:attribute name="type">coverart</xsl:attribute>
-                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com/banners/', normalize-space(poster))"/></xsl:attribute>
-                                <xsl:attribute name="thumb"><xsl:value-of select="concat('http://www.thetvdb.com/banners/_cache/', normalize-space(poster))"/></xsl:attribute>
+                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com', normalize-space(poster))"/></xsl:attribute>
+                                <xsl:attribute name="thumb"><xsl:value-of select="tvdbXpath:replace(concat('http://www.thetvdb.com', normalize-space(poster)), '/banners/', '/banners/_cache/')"/></xsl:attribute>
                             </xsl:element>
                         </xsl:if>
                         <xsl:if test=".//fanart/text() != ''">
                             <xsl:element name="image">
                                 <xsl:attribute name="type">fanart</xsl:attribute>
-                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com/banners/', normalize-space(fanart))"/></xsl:attribute>
-                                <xsl:attribute name="thumb"><xsl:value-of select="concat('http://www.thetvdb.com/banners/_cache/', normalize-space(fanart))"/></xsl:attribute>
+                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com', normalize-space(fanart))"/></xsl:attribute>
+                                <xsl:attribute name="thumb"><xsl:value-of select="tvdbXpath:replace(concat('http://www.thetvdb.com', normalize-space(fanart)), '/banners/', '/banners/_cache/')"/></xsl:attribute>
                             </xsl:element>
                         </xsl:if>
                         <xsl:if test=".//banner/text() != ''">
                             <xsl:element name="image">
                                 <xsl:attribute name="type">banner</xsl:attribute>
-                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com/banners/', normalize-space(banner))"/></xsl:attribute>
-                                <xsl:attribute name="thumb"><xsl:value-of select="concat('http://www.thetvdb.com/banners/_cache/', normalize-space(banner))"/></xsl:attribute>
+                                <xsl:attribute name="url"><xsl:value-of select="concat('http://www.thetvdb.com', normalize-space(banner))"/></xsl:attribute>
+                                <xsl:attribute name="thumb"><xsl:value-of select="tvdbXpath:replace(concat('http://www.thetvdb.com', normalize-space(banner)), '/banners/', '/banners/_cache/')"/></xsl:attribute>
                             </xsl:element>
                         </xsl:if>
                     </images>
